lib/LedgerSMB/DBObject/Reconciliation.pm

```diff
@@ -162,7 +162,9 @@ sub approve {
     # the user should be embedded into the $self object.
     my $report_id = shift @_;
 
-    my $code = $self->call_dbmethod(funcname=>'reconciliation__report_approve', args=>[$report_id]); # user
+    # Even with https://github.com/ledgersmb/LedgerSMB/issues/838 fixes, array parameters DO NOT WORK
+    #my $code = $self->call_dbmethod(funcname=>'reconciliation__report_approve', args=>[$report_id]); # user
+    my $code = $self->call_dbmethod(funcname=>'reconciliation__report_approve', args=> { in_report_id => $report_id }); # user
```

Comment:
@ylavoie There might be a need to have a look at the entire module. I'll schedule that for tomorrow. Although positional parameters indeed do require tight coupling, I think that's the point here: the db function and the module and the sproc *are* tightly coupled in the sense that the module provides a Perlish wrapping interface to the stored proc.